events/api.py

```diff
@@ -1519,44 +1563,21 @@ def create(self, validated_data):
                     raise ParseError(
                         f"{org_type} should be a list, you provided {type(self.request.data[org_type])}"
                     )
-        if "parent_organization" in self.request.data.keys():
-            user = self.request.user
-            parent_id = (
-                self.request.data["parent_organization"].rstrip("/").split("/")[-1]
-            )
-            potential_parent = Organization.objects.filter(id=parent_id)
-            if potential_parent.count() == 0:
-                raise ParseError(
-                    f"{self.request.data['parent_organization']} does not exist."
-                )
-            if user:
-                if utils.organization_can_be_edited_by(potential_parent[0], user):
-                    pass
-                else:
-                    raise DRFPermissionDenied("User has no rights to this organization")
-            else:
-                raise DRFPermissionDenied(
-                    "User must be logged in to create organizations."
-                )
+
         org = super().create(validated_data)
-        self.connected_organizations(conn_orgs_in_request, org)
+        self.connect_organizations(conn_orgs_in_request, org)
+
         return org
 
-    def get_regular_users(self, obj):
-        user = self.context["user"]
-        if not isinstance(user, AnonymousUser) and user.is_admin_of(obj):
-            return UserSerializer(
-                obj.regular_users.all(), read_only=True, many=True
-            ).data
-        else:
-            return []
+    def update(self, instance, validated_data):
+        # Prevent user to accidentally remove himself from admin
+        if validated_data.get("admin_users") is not None:
+            admin_users = validated_data.pop("admin_users")
+            instance.admin_users.clear()
+            instance.admin_users.add(*admin_users)
+            instance.admin_users.add(self.user)
```

Comment:
This part can be replaced with:

```suggestion
            instance.admin_users.set([*admin_users, self.user])
```

Reply:
Changed as suggested